src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,93 @@
+
+#include <array>
+#include <boost/math/quaternion.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+/// \brief Convert a `boost::math::quaternion` to a `std::array<DataVector, 1>`
+///
+/// \details Functions of time return `std::array<DataVector, N>`s from their
+/// `func` members where `N` is the number of derivatives of the function.
+/// For quaternions, only the quaternion itself is stored so only
+/// need to create a `std::array<DataVector, 1>` to return
+template <typename T>
+std::array<DataVector, 1> quaternion_to_array_dv(
+    const boost::math::quaternion<T>& input) noexcept {
+  DataVector dv{input.R_component_1(), input.R_component_2(),
+                input.R_component_3(), input.R_component_4()};
+  return std::array<DataVector, 1>{dv};
+}
+
+/// \brief Convert a `std::array<DataVector, 1>` to a `boost::math::quaternion`
+///
+/// \details Functions of time return `std::array<DataVector, N>`s from their
+/// `func` members where `N` is the number of derivatives of the function.
+/// For quaternions, only the quaternion itself is stored so only a
+/// conversion from `std::array<DataVector, 1>` is needed.
+template <typename T>
+boost::math::quaternion<T> array_dv_to_quaternion(
+    const std::array<DataVector, 1>& input) noexcept {
+  return boost::math::quaternion<T>(
+      static_cast<T>(input[0][0]), static_cast<T>(input[0][1]),
+      static_cast<T>(input[0][2]), static_cast<T>(input[0][3]));
+}
+
+/// \brief Convert a `DataVector` to a `boost::math::quaternion`
+///
+/// \details To conver to a quaternion, a `DataVector` must have either 3 or 4
+/// components. If it has 3 components, the quaternion will be constructed with
+/// 0 scalar part while the vector part is the `DataVector`. If the `DataVector`
+/// has 4 components, the quaternion is just the `DataVector` itself.
+template <typename T>
+boost::math::quaternion<T> datavector_to_quaternion(
+    const DataVector& input) noexcept {
+  ASSERT(input.size() == 3 or input.size() == 4,
+         "To form a quaternion, a DataVector can either have 3 or 4 components "
+         "only.");
+  boost::math::quaternion<T> result;
+  if (input.size() == 3) {
+    result = boost::math::quaternion<T>(
+        static_cast<T>(0), static_cast<T>(input[0]), static_cast<T>(input[1]),
+        static_cast<T>(input[2]));
+  } else {
+    result = boost::math::quaternion<T>(
+        static_cast<T>(input[0]), static_cast<T>(input[1]),
+        static_cast<T>(input[2]), static_cast<T>(input[3]));
+  }
+  return result;
+}
+/// Normalize a `boost::math::quaternion`
+template <typename T>
+void normalize_quaternion(
+    gsl::not_null<boost::math::quaternion<T>*> input) noexcept {
+  *input /= abs(*input);
+}
+
+/// Stream operator specifically for `std::array<DataVector, Dim>`
+// needed because Test_QuaternionHelpers.cpp won't compile if this isn't defined
```

Comment:
Does including `"Utilities/StdHelpers.hpp"` in the test fix this?  This function doesn't have anything to do with quaternions.